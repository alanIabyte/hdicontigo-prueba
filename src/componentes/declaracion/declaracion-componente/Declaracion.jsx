import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import IconoCerrar from "@material-ui/icons/CloseRounded";
import {
  EnvolvedorDeclaracion,
  TextoSeguirLeyendo,
  EnvolvedorModalDeclaracion,
  ModalContenedor,
  EncabezadoModal,
  ContenedorIconoCierre,
} from "./Declaracion.styled";
import obtenerValorDeArregloDeStrings from "../../utilidades-compartidas/obtenerValorDeArregloDeStrings";

const Declaracion = (props) => {
  const { datosDeclaracion } = props;
  const [mostrarDeclaracion, asignarValorMostrarDeclaracion] = useState(false);

  const diccionario = {
    titulo: "Declaración",
    seguirLeyendo: "Seguir leyendo",
  };

  const declaracion = obtenerValorDeArregloDeStrings(
    datosDeclaracion,
    "Texto: "
  );

  // Si la longitud de la declaración es menor a 200 caracteres, mostrarla completa.
  const esDeclaracionCorta = !(declaracion.length > 200);

  useEffect(() => {
    if (mostrarDeclaracion) {
      document.body.style.overflow = "hidden";
    } else {
      document.body.style.overflow = "unset";
    }
  }, [mostrarDeclaracion]);

  const ModalDeclaracion = () => (
    <EnvolvedorModalDeclaracion mostrar={mostrarDeclaracion}>
      <ModalContenedor>
        <EncabezadoModal>
          {diccionario.titulo}
          <ContenedorIconoCierre
            onClick={() => {
              asignarValorMostrarDeclaracion(false);
            }}
          >
            <IconoCerrar />
          </ContenedorIconoCierre>
        </EncabezadoModal>
        {declaracion}
      </ModalContenedor>
    </EnvolvedorModalDeclaracion>
  );

  return (
    <EnvolvedorDeclaracion>
      <ModalDeclaracion />
      {esDeclaracionCorta ? (
        <>{declaracion}</>
      ) : (
        <>
          {declaracion.substring(0, 135)}...
          <TextoSeguirLeyendo
            onClick={() => {
              asignarValorMostrarDeclaracion(true);
            }}
          >
            {diccionario.seguirLeyendo}
          </TextoSeguirLeyendo>
        </>
      )}
    </EnvolvedorDeclaracion>
  );
};

Declaracion.propTypes = {
  datosDeclaracion: PropTypes.arrayOf(PropTypes.string),
};

Declaracion.defaultProps = {
  datosDeclaracion: [
    // eslint-disable-next-line max-len
    "Texto: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Sed blandit libero volutpat sed cras ornare arcu. Tellus orci ac auctor augue mauris augue. Interdum velit euismod in pellentesque. Viverra nam libero justo laoreet sit. Posuere ac ut consequat semper viverra nam libero justo. Convallis a cras semper auctor. Odio facilisis mauris sit amet massa vitae tortor condimentum lacinia. Nunc mi ipsum faucibus vitae aliquet nec. Duis at consectetur lorem donec. Dictum varius duis at consectetur lorem donec massa sapien. Massa vitae tortor condimentum lacinia quis vel eros donec ac. Nullam eget felis eget nunc lobortis mattis.",
  ],
};

export default Declaracion;
