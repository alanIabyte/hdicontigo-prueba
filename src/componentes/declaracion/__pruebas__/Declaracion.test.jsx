import React from "react";
import renderer from "react-test-renderer";
import Declaracion from "../declaracion-componente/Declaracion";

describe("#Declaracion", () => {
  it("se despliega correctamente", () => {
    const tree = renderer.create(<Declaracion />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
