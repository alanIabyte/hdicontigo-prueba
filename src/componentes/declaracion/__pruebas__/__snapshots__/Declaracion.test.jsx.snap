// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Declaracion se despliega correctamente 1`] = `
<div
  className="sc-iGgWBj sc-ikkxIA fgQoGq gvzrlR"
>
  <div
    className="sc-feUZmu dXvHse"
  >
    <div
      className="sc-fUnMCh gkAIpe"
    >
      <div
        className="sc-fqkvVR sc-hzhJZQ wQLUp fYDaCd"
      >
        Declaraci√≥n
        <button
          className="sc-fHjqPf lpkQKc"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.3 5.71a.9959.9959 0 00-1.41 0L12 10.59 7.11 5.7a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
            />
          </svg>
        </button>
      </div>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Sed blandit libero volutpat sed cras ornare arcu. Tellus orci ac auctor augue mauris augue. Interdum velit euismod in pellentesque. Viverra nam libero justo laoreet sit. Posuere ac ut consequat semper viverra nam libero justo. Convallis a cras semper auctor. Odio facilisis mauris sit amet massa vitae tortor condimentum lacinia. Nunc mi ipsum faucibus vitae aliquet nec. Duis at consectetur lorem donec. Dictum varius duis at consectetur lorem donec massa sapien. Massa vitae tortor condimentum lacinia quis vel eros donec ac. Nullam eget felis eget nunc lobortis mattis.
    </div>
  </div>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad 
  ...
  <div
    className="sc-kpDqfm sc-dAbbOL bTCkFB htGLfa"
    onClick={[Function]}
  >
    Seguir leyendo
  </div>
</div>
`;
